addons/product/controllers/catalog.py

```diff
@@ -25,6 +25,7 @@ def product_catalog_get_order_lines_info(self, res_model, order_id, product_ids,
             }
         """
         order = request.env[res_model].browse(order_id)
+        request.env.company = order.company_id
```

Comment:
this works but is a little hacky: the environment of `order` and `request` is shared and when we write on the company_id in the `request` env, we update the `order` env. this works here and would probably be safe, but I think it's better to set the company via the context and let the lazy_property company of env compute the company.

here we only need it for recordset `order`, so for example we could do:

```suggestion
        order = order.with_company(order.company_id)
```

if we wanted to update the request company (but here we don't need it) so it uses another company on the next usage of request, we could do:

```
request.update_context(allowed_company_ids=order_id.company_id.ids)
```
that will create a set a new environment to request object separate from other recordset.